dask/dataframe/io/parquet/arrow.py

```diff
@@ -332,6 +339,65 @@ class ArrowDatasetEngine(Engine):
     # Public Class Methods
     #
 
+    @classmethod
+    def extract_filesystem(
+        cls,
+        urlpath,
+        filesystem,
+        dataset_options,
+        open_file_options,
+        storage_options,
+    ):
+
+        # Check if filesystem was specified as a dataset option
+        fs = dataset_options.pop("filesystem", None)
+        if filesystem is not None:
+            fs = filesystem
+
+        fs = fs or "fsspec"  # Default is fsspec
+
+        # Handle pyarrow-based filesystem
+        if isinstance(fs, pa_fs.FileSystem) or fs in ("arrow", "pyarrow"):
+            if isinstance(urlpath, (list, tuple, set)):
+                if not urlpath:
+                    raise ValueError("empty urlpath sequence")
+                urlpath = [stringify_path(u) for u in urlpath]
+            else:
+                urlpath = [stringify_path(urlpath)]
+
+            if fs in ("arrow", "pyarrow"):
+                fs = type(pa_fs.FileSystem.from_uri(urlpath[0])[0])(
+                    **(storage_options or {})
+                )
+
+            if isinstance(fs, pa_fs.FileSystem):
+                fsspec_fs = ArrowFSWrapper(fs)
+                if urlpath[0].startswith("C:") and isinstance(
+                    fs, pa_fs.LocalFileSystem
+                ):
+                    # ArrowFSWrapper._strip_protocol not reliable on windows
+                    from fsspec.implementations.local import LocalFileSystem
+
+                    fs_strip = LocalFileSystem()
```

Comment:
Is there an upstream issue for this one?

Reply:
I submitted this issue and linked in a comment: https://github.com/fsspec/filesystem_spec/issues/1137